static/app/controllers.js

```diff
@@ -92,136 +92,110 @@
+        // Get all the users associated with an org
+        $scope.org_users = [];
+        $scope.loadComplete = {
+            status: false
+        };
+
+	$scope.addUserToOrg = function(user) {
             user.id = undefined;
+            user.error = undefined;
             return $uaa.getUserGuidFromUserName(user)
                 .then(function(user) {
-                    $cloudfoundry.setOrgUserCategory($routeParams['orgguid'], user.id, 'users', true)
-                        .then(function() {
-                            $scope.showTab('current_org_users');
-                        });
+                    if (!user.error) {
+                        user.metadata = {guid: user.id};
+                        user.users = true; // want it to become true.
+                        $cloudfoundry.toggleOrgUserPermissions(user, "users", $routeParams['orgguid'])
+                            .then(function() {
+                                $scope.showTab('current_org_users');
+                            });
+                    } else {
+                        $scope.inviteOrgUserError = user.error;
+                    }
                 }).catch(function(error) {
-                    console.log('Unable to add the user');
+                    $scope.inviteOrgUserError = error;
                 });
         };
-        $scope.removeUserFromOrg = function(userGuid) {
+        $scope.removeUserFromOrg = function(user) {
+            user.users = false; // want it to be false aka removed.
+            user.managers = user.entity.organization_roles.indexOf('org_manager') >= 0;
+            user.auditors = user.entity.organization_roles.indexOf('org_auditor') >= 0;
+            user.billing_managers = user.entity.organization_roles.indexOf('billing_manager') >= 0;
+            $scope.removeOrgUserError = null;
+            if (user.managers || user.auditors || user.billing_managers) {
+                $scope.removeOrgUserError = "Please remove all other organization roles for the user before attempting to remove the user from the organization";
+                return;
+            }
             // TODO: remove all roles before removing from org so that if the user is re-added, the user will have no roles.
-            return $cloudfoundry.setOrgUserCategory($routeParams['orgguid'], userGuid, 'users', false)
-                .then(function() {
-                    $scope.showTab('current_org_users')
+            return $cloudfoundry.toggleOrgUserPermissions(user, "users",$routeParams['orgguid'])
+                .then(function(response) {
+                    if (response.status !== 201) {
+                        $scope.removeOrgUserError = response.data.description;
+                    } else {
+                        $scope.showTab('current_org_users')
+                    }
                 })
         };
         // Show a specific tab
         $scope.showTab = function(tab) {
+            $scope.removeOrgUserError = null;
```

Comment:
might want to clear `$scope.inviteOrgUserError` here so that the message doesn't persist.